pkg/plugins/builtin/script/script.go

```diff
@@ -0,0 +1,147 @@
+package script
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	gexec "os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/ovh/utask"
+
+	"github.com/ovh/utask/pkg/plugins/taskplugin"
+)
+
+// the script plugin execute scripts
+var (
+	Plugin = taskplugin.New("script", "0.1", exec,
+		taskplugin.WithConfig(validConfig, Config{}),
+	)
+)
+
+// Metadata represents the metadata of script execution
+type Metadata struct {
+	ExitCode      string `json:"exit_code"`
+	ProcessState  string `json:"process_state"`
+	Output        string `json:"output"`
+	ExecutionTime string `json:"execution_time"`
+	Error         string `json:"error"`
+}
+
+// Config is the configuration needed to execute a script
+type Config struct {
+	File    string   `json:"file"`
+	Argv    []string `json:"argv,omitempty"`
+	Timeout string   `json:"timeout,omitempty"`
+	Stdin   string   `json:"stdin,omitempty"`
+}
+
+func validConfig(config interface{}) error {
+	cfg := config.(*Config)
+
+	if cfg.File == "" {
+		return errors.New("file is missing")
+	}
+
+	f, err := os.Stat(filepath.Join(utask.FScriptsFolder, cfg.File))
+	if os.IsNotExist(err) {
+		return fmt.Errorf("%s not found in FS: %s", cfg.File, err.Error())
+	}
+
+	if f.Mode()&0010 == 0 {
```

Comment:
Why checking `g+x` and not `o+x` or `u+x` ?
 
```suggestion
	if f.Mode()&0111 == 0 {
```

Will tell you if it's executable by somebody

Reply:
Because we only care if we're able to execute it based on group owner.
There is some cases you're not the main owner but the user belong to the group owner.
We also don't care about other users permissions.

Reply:
In fact, all you can check is that the script is at least executable, by someone.

Here you check that the group associated to the file can execute it, but you don't check that the user belong to that group. So the check is useless.

Here your test will fail but I can exec the script:
```
toms@thor:~/utask$ ls -alhtr a.sh
-rwxr--r-- 1 toms toms 0 Dec 15 19:36 a.sh
toms@thor:~/utask$ ./a.sh
toms@thor:~/utask$
```

Here your test will succeed but I can't exec the script:
```
toms@thor:~/utask$ ls -alhtr b.sh
-rw-r-xr-- 1 toms root 0 Dec 15 19:36 b.sh
toms@thor:~/utask$ ./b.sh
-bash: ./b.sh: Permission denied
toms@thor:~/utask$
```
Either check that the file is at least executable by someone, or do the full permission check